src/markdown-pages/install-with-kurl/migrating-csi.md

```diff
@@ -3,73 +3,206 @@ path: "/docs/install-with-kurl/migrating-csi"
+    ```
+
+    If any volumes are reported as not healthy in the `Robustness` column in the ouput of this command, resolve the issue before proceeding.
+
+    To learn more about volume health, you can also inspect each volume individually:
+
+    ```
+    kubectl get volumes.longhorn.io -n longhorn-system <volume name> -o yaml
+    ```
+
+    In many cases, volume health is caused by issues with volume replication. Specifically, when multiple replicas are configured for a volume but not all have been scheduled. 
+
+    _**Note**_: During the data migration process in single-node clusters, the system automatically scales down the number of replicas to 1 in all Longhorn volumes to ensure the volumes are in a healthy state before beginning the data transfer. This is done to minimize the risk of a migration failure.
+
+- Confirm that Longhorn nodes are in a healthy state. The nodes must be healthy to ensure they are not over-provisioned and can handle scheduled workloads. Run the following command to check the status of the Longhorn nodes:
+
+    ```
+    kubectl get nodes.longhorn.io -A
+    ```
+
+    If any node is not reported as "Ready" and "Schedulable" in the output of this command, resolve the issue before proceeding.
+    
+    To learn more, you can also inspect each node individually and view its "Status" property:
+
+    ```
+    kubectl get nodes.longhorn.io -n longhorn-system <node name> -o yaml
+    ```
+
+- (OpenEBS Only) Before you migrate from Longhorn to OpenEBS:
+   - Ensure the filesystem on the node has adequate space to accommodate twice the amount of data currently stored by Longhorn. This is important because both OpenEBS and Longhorn use the node's filesystem for data storage.
+   - Ensure that there is an additional 2G of memory and 2 CPUs available for OpenEBS. For more information, see [What are the minimum requirements and supported container orchestrators?](https://openebs.io/docs/faqs/general#what-are-the-minimum-requirements-and-supported-container-orchestrators) in the OpenEBS documentation.
+
+- (Rook Only) Before you migrate from Longhorn to Rook, ensure that the dedicated block device for Rook attached to each node has enough space to host all data currently stored in Longhorn.
+
+## Change the CSI Add-on in a Cluster
+
+This procedure describes how to update the kURL specification file to use a new CSI provisioner add-on. Then, upgrade your kURL cluster to automatically migrate data to the new provisioner.
+
+For more information about the supported migration paths for CSI provisioners, see [Supported CSI Migrations](#supported-csi-migrations) above.
+
+_**Warning**_: When you change the CSI provisioner in your cluster, the data migration process causes some amount of downtime for the application. It is important to plan accordingly to minimize the impact on users.
+
+To migrate to a new CSI provisioner in a kURL cluster:
+
+1. Complete the [Prerequisites](#prerequisites) above.
```

Comment:
LMK if you all would like to stronger language in any parts of this procedure. Not sure how "warning-y" we want to be.